apps/dashboard/app/models/announcement.rb

```diff
@@ -31,10 +32,36 @@ def msg
     msg.blank? ? nil : msg
   end
 
+  def button_text
+    opts.fetch(:button_text, I18n.t('dashboard.announcements_button_text')).to_s
+  end
+
   # Whether this is a valid announcement
   # @return [Boolean] whether it is valid
   def valid?
-    !!msg
+    return false unless msg
+
+    return false if dismissible? && !id
+
+    true
+  end
+
+  def completed?
+    dismissible? && user_settings.dig(:announcements, id.to_sym)
+  end
+
+  def dismissible?
+    required? || opts.fetch(:dismissible, false)
+  end
```

Comment:
I'm wondering if all announcements should be dismissable. I'm thinking yes as they take up a bit of real estate and can be on all pages for days. 

I guess as a best of both worlds - we can just default here to `true` and folks can set `false` if they wish to keep it on the page.

Reply:
Fixed